src/Splitio/Services/Client/Classes/SelfRefreshingClient.cs

```diff
@@ -199,23 +206,28 @@ private void BuildBlockUntilReadyService()
         private void BuildTelemetrySyncTask()
         {
             var httpClient = new SplitioHttpClient(ApiKey, _config, GetHeaders());
+            var statsTask =  _tasksManager.NewPeriodicTask(Enums.Task.TelemetryStats, _config.TelemetryRefreshRate * 1000);
+            var initTask = _tasksManager.NewOnTimeTask(Enums.Task.TelemetryInit);
```

Comment:
do we need a task for Init? Why? is executed only once

Reply:
to execute in parallel and not on the main thread.